osu.Game/Screens/OnlinePlay/FooterButtonFreeMods.cs

```diff
@@ -49,14 +87,42 @@ protected override void LoadComplete()
             base.LoadComplete();
 
             Current.BindValueChanged(_ => updateModDisplay(), true);
+
+            // Overwrite any external behaviour as we delegate the main toggle action to a sub-button.
+            Action = toggleAllFreeMods;
+        }
+
+        /// <summary>
+        /// Immediately toggle all free mods on/off.
+        /// </summary>
+        private void toggleAllFreeMods()
+        {
+            var availableMods = freeModSelectOverlay.AllAvailableAndValidMods.ToArray();
+
+            Current.Value = Current.Value.Count == availableMods.Length
+                ? Array.Empty<Mod>()
+                : availableMods;
         }
 
         private void updateModDisplay()
         {
-            if (Current.Value?.Count > 0)
-                modDisplay.FadeIn();
+            int current = Current.Value.Count;
+
+            if (current == freeModSelectOverlay.AllAvailableAndValidMods.Count())
+            {
+                count.Text = "all";
+                circle.FadeColour(colours.Yellow, 200, Easing.OutQuint);
+            }
+            else if (current > 0)
+            {
+                count.Text = $"{current} mods";
+                circle.FadeColour(colours.YellowDark, 200, Easing.OutQuint);
+            }
```

Comment:
The foreground-to-background contrast in the "some mods" state is passable, but in the "all" state is pretty dreadful. Propose inverting foreground colour.

```diff
diff --git a/osu.Game/Screens/OnlinePlay/FooterButtonFreeMods.cs b/osu.Game/Screens/OnlinePlay/FooterButtonFreeMods.cs
index 56a69be741..897e7c39c7 100644
--- a/osu.Game/Screens/OnlinePlay/FooterButtonFreeMods.cs
+++ b/osu.Game/Screens/OnlinePlay/FooterButtonFreeMods.cs
@@ -111,16 +111,19 @@ private void updateModDisplay()
             if (current == freeModSelectOverlay.AllAvailableAndValidMods.Count())
             {
                 count.Text = "all";
+                count.FadeColour(colours.Gray2, 200, Easing.OutQuint);
                 circle.FadeColour(colours.Yellow, 200, Easing.OutQuint);
             }
             else if (current > 0)
             {
                 count.Text = $"{current} mods";
+                count.FadeColour(colours.Gray2, 200, Easing.OutQuint);
                 circle.FadeColour(colours.YellowDark, 200, Easing.OutQuint);
             }
             else
             {
                 count.Text = "off";
+                count.FadeColour(colours.GrayF, 200, Easing.OutQuint);
                 circle.FadeColour(colours.Gray4, 200, Easing.OutQuint);
             }
         }

```

https://github.com/ppy/osu/assets/20418176/9db1c206-a242-4cbd-a994-c9072f38273c